lib/cli/interactive-setup/console-dev-mode-feed.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const { writeText, style } = require('@serverless/utils/log');
+const WebSocket = require('ws');
+const chalk = require('chalk');
+const { devModeFeed } = require('@serverless/utils/lib/auth/urls');
+const consoleUi = require('@serverless/utils/console-ui');
+const resolveAuthMode = require('@serverless/utils/auth/resolve-mode');
+const streamBuffers = require('stream-buffers');
+const apiRequest = require('@serverless/utils/api-request');
+const promptWithHistory = require('@serverless/utils/inquirer/prompt-with-history');
+
+const streamBuff = new streamBuffers.ReadableStreamBuffer({
+  frequency: 500,
+  chunkSize: 2048 * 1000000,
+});
+
+const consoleMonitoringCounter = {
+  logBatches: 0,
+  events: 0,
+  responses: 0,
+};
+
+const handleSocketMessage = (data) => {
+  try {
+    const splitData = data.toString('utf-8').split(';;;');
+    const jsonArray = splitData.reduce((arr, item) => {
+      try {
+        const parsedItem = JSON.parse(item);
+        if (Array.isArray(parsedItem)) {
+          return [...arr, ...parsedItem];
+        }
+        throw new Error('Not an array');
+      } catch (error) {
+        return arr;
+      }
+    }, []);
+    if (!Array.isArray(jsonArray)) return;
```

Comment:
Updated to avoid `reduce` 😎 Note we need to check the array because when we first establish a connection with filters applied or not the websocket sends a successful connection message that is not an array.

Reply:
Okay never mind I couldn't get the flat map thing to work in all scenarios for some reason where as the reduce is working 🤷

Reply:
Ignore my deleted comment 😆 It wasn't the flat map thing.